src/Sulu/Component/Content/Document/Subscriber/DocumentSynchronizationSubscriber.php

```diff
@@ -0,0 +1,213 @@
+use Sulu\Component\DocumentManager\Events;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class DocumentSynchronizationSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var SynchronizationManager
+     */
+    private $syncManager;
+
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $defaultManager;
+
+    /**
+     * @var object[]
+     */
+    private $persistQueue = [];
+
+    /**
+     * @var object[]
+     */
+    private $removeQueue = [];
+
+    /**
+     * NOTE: We pass the default manager here because we need to ensure that we
+     *       only process documents FROM the default manager. If we could assign
+     *       event subscribers to specific document managers this would not
+     *       be necessary.
+     */
+    public function __construct(DocumentManagerInterface $defaultManager, SynchronizationManager $syncManager)
+    {
+        $this->defaultManager = $defaultManager;
+        $this->syncManager = $syncManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            Events::FLUSH => 'handleFlush',
+            Events::REMOVE => 'handleRemove',
+            Events::METADATA_LOAD => 'handleMetadataLoad',
+
+            // persist needs to be before the content mapper subscriber
+            // because we need to stop propagation early on the publish
+            Events::PERSIST => ['handlePersist', 10],
+        ];
+    }
+
+    public function handleMetadataLoad(MetadataLoadEvent $event)
+    {
+        $metadata = $event->getMetadata();
+
+        if (false === $metadata->getReflectionClass()->isSubclassOf(SynchronizeBehavior::class)) {
+            return;
+        }
+
+        $encoding = $metadata->getReflectionClass()->isSubclassOf(LocaleBehavior::class) ? 'system_localized' : 'system';
+
+        $metadata->addFieldMapping('synchronizedManagers', [
+            'encoding' => $encoding,
+            'property' => SynchronizeBehavior::SYNCED_FIELD,
+            'type' => 'string',
+        ]);
+    }
+
+    /**
+     * Synchronize new documents with the publish document manager.
+     *
+     * @param PersistEvent
+     */
+    public function handlePersist(PersistEvent $event)
+    {
+        $manager = $event->getManager();
+
+        // do not do anything if the default and publish managers are the same.
+        if ($this->defaultManager === $this->syncManager->getPublishDocumentManager()) {
+            return;
+        }
+
+        $this->assertEmittingManagerDefaultManager($manager);
+
+        $document = $event->getDocument();
+
+        // only sync documents implementing the sync behavior
+        if (!$document instanceof SynchronizeBehavior) {
+            return;
+        }
+
+        // only sync new documents automatically
+        if (false === $event->getNode()->isNew()) {
+            // otherwise the node is now capable of synchronized with all the managers.
```

Comment:
This command is a bit confusing... What's happening here? The value of `synchronizedManagers` is set to an empty array, and this is called whenever a node is edited, right?